README.md

```diff
@@ -1,222 +1,515 @@
+  }
+
+  fetchFoo: function(){
+    // The message is assigned to a local variable
+    const message = this.message
+
+    const getMessage = function(){
+      return message;
+    };
+
+    return fetch('some/ajax/request')
+      .then(getMessage);
+  }
+}
+```
+
+### Curly Braces
+
+Curly braces are required around blocks in loops and conditionals.
 
 **Don't do this:**
+
 ```javascript
-var x;  //x is never used
+if (foo)
+  console.log('bar');
 ```
-**But this is okay:**
+
+**Do this instead:**
+
 ```javascript
-define(function(require, exports, module){
-  //module is never used
-});
+if (foo) {
+  console.log('bar');
+}
 ```
 
-## Coerced Types and Equality
+### Strict Mode
+
+* All modules should be have strict mode enabled.
+* ES6 modules have strict mode enabled by default.
+* Non-ES6 modules should have `'use strict';` placed at the top of the file.
+
+### Coerced Types and Equality
 
-Prefer `===` over `==`. This will be [enforced by JShint](http://www.jshint.com/docs/options/#eqeqeq).
+Strict comparison (`===`) should be used instead of abstract comparison (`==`).
 
-## Undefined Variables
+### Trailing Whitespace
 
-All variables must be explicitly defined. This option will be [enforced by JSHint](http://www.jshint.com/docs/options/#undef).
+Lines shouldn't have whitespace at the end.
 
-This is a simple way to catch mistyped variable names and copy/paste errors, but it also helps to prevent over-dependence on the global namespace, and forces modules to be more explicit about their dependencies.
+### 80 Column Limit
 
-One specific example of why this can be important is in Backbone views, which use a scoped `$` function to find elements inside the view. If you accidentally use the global `$('.selector')` instead of `this.$('.selector')`, you're searching the full DOM instead of just your view.
+Lines should not exceed 80 characters.
 
-## Using Variables Before Definition
+In specific cases, long lines can be ignored by the linter.
+
+**Don't do this:**
 
-Variables must be defined earlier in the code than they are used. (Hoisting makes it possible to accidentally use a variable before it's defined.) This will be [enforced by JSHint](http://www.jshint.com/docs/options/#latedef).
+```javascript
+let longTestPermalinkUrl = 'https://example.com/v1/en/2016/03/14/assets/images/1234.jpg';
+```
 
-## Trailing Whitespace
+**Do this instead:**
 
-Lines shouldn't have whitespace at the end. We've always enforced this. Now [JSHint will enforce it](http://www.jshint.com/docs/options/#trailing).
+```javascript
+let longTestPermalinkUrl;
+longTestPermalinkUrl = 'https://example.com/v1/en/2016/03/14/assets/images/1234.jpg'; // jshint ignore:line
+```
 
-## 80 Column Limit
+## License
 
-Although the style guide isn't fully explicit about it, lines should not exceed 80 characters.
+Copyright &copy; 2016 [Yola](http://yola.com).
+<br>Released under the [MIT License](https://github.com/yola/jsgreat/blob/master/LICENSE).
```

Comment:
It's better to be clear, and call it the MIT/Expat license. https://en.wikipedia.org/wiki/MIT_License#Various_versions